server/mdm/nanomdm/service/nanomdm/service.go

```diff
@@ -232,7 +232,19 @@ func (s *Service) CommandAndReportResults(r *mdm.Request, results *mdm.CommandRe
 	logger.Info(logs...)
 	err := s.store.StoreCommandReport(r, results)
 	if err != nil {
-		return nil, fmt.Errorf("storing command report: %w", err)
+		// allow not found commands, this is an edge case only
+		// valid for migrations, other response codes confuse the
+		// mdmclient, and this gives us the opportunity to answer
+		// with more commands
```

Comment:
What happens with other response codes? Does the device keep trying to report the results for the not found command each checkin so we never get to send the next command?

Reply:
yes, exactly